sfge/src/test/java/com/salesforce/rules/UnimplementedTypeRuleTest.java

```diff
@@ -0,0 +1,253 @@
+package com.salesforce.rules;
+
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.salesforce.TestUtil;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+import org.hamcrest.MatcherAssert;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class UnimplementedTypeRuleTest {
+    private GraphTraversalSource g;
+
+    @BeforeEach
+    public void setup() {
+        this.g = TestUtil.getGraph();
+    }
+
+    @MethodSource("paramProvider_testOuterTypes")
+    @ParameterizedTest(name = "{displayName}: Tested type = {0}")
+    public void testOuterTypes(
+            String inheriteeDef,
+            String inheritorDef,
+            String inheritance,
+            List<String> expectedTypes) {
+        // This type is unimplemented.
+        String unextendedSource = inheriteeDef + " UnextendedType {}";
+        // This type is implemented.
+        String extendedSource = inheriteeDef + " ExtendedType {}";
+        // This type does the implementing.
+        String extenderSource = inheritorDef + " ExtenderType " + inheritance + " ExtendedType {}";
+
+        executeTest(expectedTypes, extendedSource, unextendedSource, extenderSource);
+    }
+
+    @MethodSource("paramProvider_testInnerTypes")
+    @ParameterizedTest(name = "{displayName}: Tested type = {0}")
+    public void testInnerTypes(
+            String inheriteeDef,
+            String inheritorDef,
+            String inheritance,
+            List<String> expectedTypes) {
+        // spotless:off
```

Comment:
This is why I added the change to the gradle build file. It's helpful for the mocked up code to actually look like code.